pkg/gadgets/process-collector/tracer/tracer.go

```diff
@@ -60,29 +60,48 @@ func RunCollector(mntnsmap string) (string, error) {
 		},
 	)
 	if err != nil {
-		return "", fmt.Errorf("failed to create BPF collection: %w", err)
+		return nil, fmt.Errorf("failed to create BPF collection: %w", err)
 	}
 
 	dumpTask, ok := coll.Programs[BPF_ITER_NAME]
 	if !ok {
-		return "", fmt.Errorf("failed to find BPF iterator %q", BPF_ITER_NAME)
+		return nil, fmt.Errorf("failed to find BPF iterator %q", BPF_ITER_NAME)
 	}
 	dumpTaskIter, err := link.AttachIter(link.IterOptions{
 		Program: dumpTask,
 	})
 	if err != nil {
-		return "", fmt.Errorf("failed to attach BPF iterator: %w", err)
+		return nil, fmt.Errorf("failed to attach BPF iterator: %w", err)
 	}
 
 	file, err := dumpTaskIter.Open()
 	if err != nil {
-		return "", fmt.Errorf("failed to open BPF iterator: %w", err)
+		return nil, fmt.Errorf("failed to open BPF iterator: %w", err)
 	}
 	defer file.Close()
 
-	contents, err := ioutil.ReadAll(file)
-	if err != nil {
-		return "", fmt.Errorf("failed to read BPF iterator: %w", err)
+	var events []processcollectortypes.Event
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		var command string
+		var tgid, pid int
+		var mntnsid uint64
+
+		line := scanner.Text()
+		fmt.Printf("line: %s\n", line)
+		matchedElems, err := fmt.Sscanf(line, "%d %d %s %d", &tgid, &pid, &command, &mntnsid)
+		if err != nil || matchedElems != 4 {
+			return nil, fmt.Errorf("failed to parse process information: %w", err)
+		}
+
+		events = append(events, processcollectortypes.Event{
+			Tgid:    tgid,
+			Pid:     pid,
+			Command: command,
+			MntNsId: mntnsid,
+		})
```

Comment:
You could fill the event right here if you pass the resolver interface to the collector as I mentioned in the other comment.